deeppavlov/models/kbqa/query_generator.py

```diff
@@ -0,0 +1,597 @@
+
+        scores_2 = self.rel_ranker([question], self.rank_list_1, False) if self.return_answers \
+                        else self.rel_ranker(question, self.rank_list_1)
+        top_rels_2 = [score[0] for score in scores_2]
+        if self.debug:
+            log.debug(f"top scored second rels: {top_rels_2}")
+
+        candidate_outputs = []
+
+        if len(entity_ids) > 1:
+            ent_combs = self.make_entity_combs(entity_ids)
+            candidate_outputs = self.find_relevant_subgraph_maxmin_two(ent_combs, top_rels_1[:self.rels_to_leave],
+                                                                       top_rels_2[:self.rels_to_leave])
+
+            ascending = self.asc_desc(question)
+            reverse = False
+            if ascending:
+                reverse = True
+            candidate_outputs = sorted(candidate_outputs, key=lambda x: x[3], reverse=reverse)
+        candidate_outputs = [(output[0], output[1], output[2]) for output in candidate_outputs]
+        if candidate_outputs:
+            candidate_outputs = [candidate_outputs[0]]
+
+        return candidate_outputs
+
+    def two_hop_solver(self, question: str,
+                       entity_ids: List[List[str]],
+                       rels_from_template: List[Tuple[str]] = None):
+        candidate_outputs = []
+        if len(entity_ids) == 1:
+            if rels_from_template is not None:
+                candidate_outputs = self.from_template_one_ent(entity_ids, rels_from_template)
+
+            else:
+                ex_rels = []
+                for entity in entity_ids[0][:self.entities_to_leave]:
+                    ex_rels += self.wiki_parser("rels", "forw", entity, type_of_rel="direct")
+                    ex_rels += self.wiki_parser("rels", "backw", entity, type_of_rel="direct")
+
+                ex_rels = list(set(ex_rels))
+                scores = self.rel_ranker([question], ex_rels, False) if self.return_answers \
+                                    else self.rel_ranker(question, ex_rels)
+                top_rels = [score[0] for score in scores]
+                if self.debug:
+                    log.debug(f"top scored rels: {top_rels}")
+
+                ex_rels_2 = []
+                for entity in entity_ids[0][:self.entities_to_leave]:
+                    for rel in top_rels[:self.rels_to_leave]:
+                        objects_mid = self.wiki_parser("objects", "forw", entity, rel, type_of_rel="direct")
+                        objects_mid += self.wiki_parser("objects", "backw", entity, rel, type_of_rel="direct")
+                        if len(objects_mid) < 10:
+                            for obj in objects_mid:
+                                ex_rels_2 += self.wiki_parser("rels", "forw", obj, type_of_rel="direct")
+
+                ex_rels_2 = list(set(ex_rels_2))
+                scores_2 = self.rel_ranker([question], ex_rels_2, False) if self.return_answers \
```

Comment:
`RelRankerInfer.__call__` method receives only two arguments, `False` is unexpected.